crates/uv-auth/src/middleware.rs

```diff
@@ -48,6 +49,54 @@ impl NetrcMode {
     }
 }
 
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq)]
+pub enum AuthPolicy {
+    /// Attempt to authenticate if needed.
+    #[default]
+    Auto,
+    /// Always attempt to authenticate.
+    Always,
+    /// Never attempt to authenticate.
+    Never,
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub struct UrlAuthPolicies(FxHashMap<Url, AuthPolicy>);
+
+impl UrlAuthPolicies {
+    pub fn new() -> Self {
+        Self(FxHashMap::default())
+    }
+
+    /// Create a new `UrlAuthPolicies` from a list of URL and `AuthPolicy`
+    /// tuples.
+    pub fn from_tuples(tuples: &[(Url, AuthPolicy)]) -> Self {
+        let mut auth_policies = UrlAuthPolicies::new();
+        for (url, auth_policy) in tuples {
+            auth_policies.add_policy(url.clone(), *auth_policy);
+        }
+        auth_policies
+    }
+
+    /// An an `AuthPolicy` for a URL.
+    pub fn add_policy(&mut self, url: Url, auth_policy: AuthPolicy) {
+        self.0.insert(url, auth_policy);
+    }
+
+    /// Get the `AuthPolicy` for a URL.
```

Comment:
Nit: wrap `AuthPolicy` in brackets? Same for all the other references. That way, editors will enable go-to definition for those references.

![Screenshot 2025-03-03 at 12 44 12 PM](https://github.com/user-attachments/assets/b72b308a-4536-4b45-9661-dc15107404e7)
![Screenshot 2025-03-03 at 12 44 22 PM](https://github.com/user-attachments/assets/a311237c-b4ef-4135-8849-f871b0dc0fc9)